### PR TITLE
fix fieldState error type with NestedValue

### LINKED ISSUE #6653: [Bug]: FieldState errors wrong typings
### Version Number

7.16.1

### Codesandbox/Expo snack

Latest version 7.16.1 does not seem to install on Codesandbox, see example repo instead

### Steps to reproduce

1. Clone example repo: https://github.com/StampixSMO/react-hook-form-fieldstate-errors-type
2. Install dependencies
3. Run `npm run compile`
4. Watch the TS error.

### Expected behaviour

The type of the errors object should not reflect the nested objects but only expose the `message` property.
This works fine in 7.15.4.

### Relevant log output

```
error TS2339: Property 'message' does
not exist on type 'DeepMap<DeepPartial<NestedValue<PhoneNumber>>, FieldError>'.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bluebill1049**
revert back to this commit: 3985e5bc308b473534c8d44621500c5ad5e4b0d5 and looks like fix the issue. @kotarella1110 I am sure you have better solution than mine 🤣

### CODE REVIEW

File: src/__tests__/controller.test.tsx [hunk 1]
```diff
@@ -1193,11 +1193,17 @@ describe('Controller', () => {
         test: string;
       };
       todos: string[];
+      nestedValue: NestedValue<{ test: string }>;
```
Comment (bluebill1049) [inline src/__tests__/controller.test.tsx:1196]: include test for `NestedValue` as well

File: src/types/controller.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {
 } from './';
 
 type ControllerFieldError<T> = DeepMapImpl<
-  DeepPartial<UnionLike<T>>,
+  DeepPartialImpl<UnionLike<T>>,
```
Comment (bluebill1049) [inline src/types/controller.ts:19]: Awesome!

**Comment by bluebill1049** [comment]
Thank you @kotarella1110!

File: src/types/utils.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export type LiteralUnion<T extends U, U extends Primitive> =
   | T
   | (U & { _?: never });
 
-type DeepPartialImpl<T> = T extends NestedValue
+export type DeepPartialImpl<T> = T extends NestedValue
```
Comment (barrymay) [inline src/types/utils.ts:21]: this is not a concern but just a curiosity (I need to spend some time wrapping my head around all the latest typings) - is there a reason that NestedValue is in the form.ts file?

Comment (bluebill1049) [reply src/types/utils.ts:21]: Sounds good to me, i tried to break those type into categories so they are a bit easier to manage.

Comment (bluebill1049) [reply src/types/utils.ts:21]: by the way, would you like to add yourself to this page? https://react-hook-form.com/about-us

Comment (barrymay) [reply src/types/utils.ts:21]: can do!

**Pull request merged**